File: 1305-all-elements-in-two-binary-search-trees/1305-all-elements-in-two-binary-search-trees.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public void traverse(PriorityQueue<Integer>pq,TreeNode n1)
    {
        if(n1==null)
        {
            return;
        }
        if(n1.left!=null)
        traverse(pq,n1.left);
        pq.add(n1.val);
        if(n1.right!=null)
        traverse(pq,n1.right);
    }
    public List<Integer> getAllElements(TreeNode root1, TreeNode root2) {
        List<Integer>res = new ArrayList<Integer>();
        PriorityQueue<Integer>pq=new PriorityQueue<>();
        traverse(pq,root1);
        traverse(pq,root2);
        while(pq.size()!=0)
        {
            res.add(pq.peek());
            pq.remove();
        }
        return res;
            
    }
}

File: 1305-all-elements-in-two-binary-search-trees/1305-all-elements-in-two-binary-search-trees.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
    
    private:
    void traverse(vector<int>&v,TreeNode* n1)
    {
  if(!n1)
        {
            return;
        }
        v.push_back(n1->val);
        cout<<n1->val;
        if(n1->left)
        traverse(v,n1->left);

        if(n1->right)
        traverse(v,n1->right);
    }
public:
    vector<int> getAllElements(TreeNode* root1, TreeNode* root2) {
        vector<int>pq;
        traverse(pq,root1);
        traverse(pq,root2);
        sort(pq.begin(),pq.end());
        return pq;
    }
};